### PR TITLE
feat: support set property

### PR DESCRIPTION
**Pull request by bzp2010**
## checklist
- [x] code style
- [x] key comments

### CODE REVIEW

**Comment by bzp2010** [comment]
## Update
I found a bug in the `proxy-wasm-go-sdk` project that will break our set property functionality. I submitted issue and the associated fix PR, but based on past commit processing and release, it is not possible to give an exact fix time.

**Issue:** https://github.com/tetratelabs/proxy-wasm-go-sdk/issues/234
**PR:** https://github.com/tetratelabs/proxy-wasm-go-sdk/pull/235

The current feature development and test code writing is done, should we wait for the `proxy-wasm-go-sdk` project developers to merge the fixes and release a new version before continuing this PR?

ping @spacewander

**Comment by bzp2010** [comment]
## Update
PR: [https://github.com/tetratelabs/proxy-wasm-go-sdk/pull/235](https://github.com/tetratelabs/proxy-wasm-go-sdk/pull/235) has been merged into the project by the maintainer and is trying to find out when the next release containing this fix will be.

**Comment by bzp2010** [comment]
## Update

The compilation problem was due to the use of `sudo make`, which caused an exception in `GOROOT` environment variable, so tinygo automatically selected Golang version `1.15.15` in actions VM, causing a compilation error, which I have fixed.

File: src/http/ngx_http_wasm_api.c
Comment (spacewander) [inline src/http/ngx_http_wasm_api.c:397]: We can save lots of code by calling ngx_http_lua_ffi_var_set directly? See https://github.com/openresty/lua-resty-core/blob/master/lib/resty/core/var.lua#L134

Comment (bzp2010) [reply src/http/ngx_http_wasm_api.c:397]: Yes, direct reuse is the best way, I will try it out.

Comment (bzp2010) [reply src/http/ngx_http_wasm_api.c]: changed

**Review by spacewander — COMMENTED** [review]
Let's copy the `proxy_set_property` from the spec to https://github.com/api7/wasm-nginx-module/blob/main/proxy_wasm_abi.md#proxy_get_property

File: src/http/ngx_http_wasm_api.c
Comment (bzp2010) [reply src/http/ngx_http_wasm_api.c:393]: Sorry, I used palloc for debugging purposes and forgot to change it back.

Comment (bzp2010) [reply src/http/ngx_http_wasm_api.c:393]: changed

File: src/http/ngx_http_wasm_api.c
Comment (bzp2010) [reply src/http/ngx_http_wasm_api.c:382]: changed

File: src/http/ngx_http_wasm_api.c
Comment (bzp2010) [reply src/http/ngx_http_wasm_api.c:398]: changed

File: src/http/ngx_http_wasm_api.c
Comment (bzp2010) [reply src/http/ngx_http_wasm_api.c:405]: changed

File: t/property.t
Comment (bzp2010) [reply t/property.t]: cleaned

**Review by spacewander — APPROVED** [review]

File: proxy_wasm_abi.md
Comment (spacewander) [inline proxy_wasm_abi.md:272]:
```suggestion
### `proxy_set_property`
```

**Pull request merged**
